scripts/executors/src/deprecate-react-components-preview-packages.ts

```diff
@@ -0,0 +1,115 @@
+import { execSync } from 'node:child_process';
+import { readdirSync } from 'node:fs';
+import { join } from 'node:path';
+
+import { getAllPackageInfo, workspaceRoot } from '@fluentui/scripts-monorepo';
+import { logger, readJsonFile } from '@nx/devkit';
+import type { ChangeType } from 'beachball';
+import yargs from 'yargs';
+
+/**
+ * Deprecates a package by executing an `npm deprecate` command.
+ *
+ * @param packageSpec - The NPM package name specifier (could be with or without a scope, and optionally tag, version, or version range, for more details see https://docs.npmjs.com/cli/v10/using-npm/package-spec)
+ * @param npmToken - The npm authentication token.
+ */
+export function deprecatePackage(packageSpec: string, npmToken: string) {
+  const command = `npm deprecate ${packageSpec} "Deprecated in favor of stable release" --registry https://registry.npmjs.org/ --//registry.npmjs.org/:_authToken=${npmToken}`;
+
+  logger.log(`Deprecating "${packageSpec}" package`);
+  logger.log(command);
+
+  try {
+    execSync(command, { stdio: 'inherit' });
+  } catch (e) {
+    throw new Error(`Failed to deprecate "${packageSpec}" package`);
+  }
+}
+
+/**
+ * Retrieves the list of v9 preview packages to deprecate based on the following criteria:
+ *   - The package has a `vNext` tag
+ *   - The package does not have a `-preview` suffix in its name
+ *   - The package version is `9.0.0-alpha.0`
+ *   - There is a change file for the package with a `minor` type
+ *
+ * @returns {string[]} An array of package names to deprecate.
```

Comment:
we are in typescript - no need for JSDoc types (this actually trigger TS warning 
<img width="455" alt="image" src="https://github.com/user-attachments/assets/f64aab62-8ab5-463a-a051-b58d29ecb714">
 

```suggestion
 * @returns An array of package names to deprecate.
```